evaluator.js

```diff
@@ -681,7 +689,7 @@ function assign_name_value(name, val, env) {
                 ? env_loop(
                     enclosing_environment(env))
                 : name === head(names)
-                  ? ( tail(head(vals))
+                  ? (tail(head(vals)) // the name should not have been declared as a constant
```

Comment:
What do you mean by `the name should not have been declared as a constant` ?

Reply:
`tail(head(vals))` tells us if the name was originally declared as a constant (if `false`) or a variable (if `true`). I think the comment would be more appropriate for the line on which the error is thrown (`no assignment to constants allowed`).

Reply:
Ok, I understand the meaning of the comment now.
Would it be clearer if we changed it to `the name was declared as a constant` ?
To me, it seems more intuitive to phrase it this way.

Reply:
Yeah the would be clearer.